### PR TITLE
FastAPI backend, fixed debounce, public, embed

### PR DESCRIPTION
**Pull request by rohanshiva**

### CODE REVIEW

File: static/public.html [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+        var parts = window.location.pathname.split("/");
+        var lastSegment = parts.pop() || parts.pop();
+        const req = await fetch(`/public/raw/${lastSegment}`, {
+          method: "GET",
+        });
+        const data = await req.json();
+        console.log(data);
+        if (data.length === 0) {
+          // no notebooks
+          dispatch(options.CreateWrapper(options.state, true));
```
Comment (xeust) [inline static/public.html:87]: we don't need to pass a stale `state` into `options` and `dispatch`, this pattern was me not using hyperapp correctly 😅

File: static/public.html [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+          let output = document.getElementById("output");
+          window.calque = new Calque(input, output);
+          input.focus();
+        });
+      };
+
+      const loadNbRecords = async (dispatch, options) => {
+        var parts = window.location.pathname.split("/");
+        var lastSegment = parts.pop() || parts.pop();
+        const req = await fetch(`/public/raw/${lastSegment}`, {
```
Comment (xeust) [inline static/public.html:80]: Curious on the implementation here?

What does `/public/raw/:segment` do?

**Review by xeust — COMMENTED** [review]
I couldn't find the `create`, `update`, and `delete` effects on the frontend for notebooks?

I see they were added to the backend but e.g. `updateNbRecord` is dispatched but couldn't find the function itself.

File: static/index.html [hunk 3]
```diff
@@ -1,359 +1,559 @@
...
+        requestAnimationFrame(() => {
+          let input = document.getElementById("input");
+          let output = document.getElementById("output");
+          window.calque = new Calque(input, output);
+          input.focus();
+        });
+      };
+      // db fx
+
+      const createNbRecord = async (dispatch, options) => {
```
Comment (rohanshiva) [inline static/index.html:79]: `createNbRecord`

File: static/index.html [hunk 4]
```diff
@@ -1,359 +1,559 @@
...
-            return resp;
+        const resp = await req.json();
+        dispatch(options.CreateNotebook(options.state, resp || {}));
+        if (options.first) {
+          // case of no notebooks, need to load Calque engine correcly
+          attachCalq(dispatch, options);
         }
+      };
+
+      const deleteNbRecord = async (dispatch, options) => {
```
Comment (rohanshiva) [inline static/index.html:105]: `deleteNbRecord`

File: static/index.html [hunk 5]
```diff
@@ -1,359 +1,559 @@
...
-            const { value: data } = await nbDB.fetch([]).next();
-            if (data.length === 0) {
-                // no notebooks
-                dispatch(options.CreateWrapper(options.state, true))
-            } else {
-                dispatch(options.ImportNotebooks(options.state, data || []))
-            }
-            return;
-        }
+      const updateNbRecord = async (dispatch, options) => {
```
Comment (rohanshiva) [inline static/index.html:118]: `updateNbRecord`

File: static/index.html [hunk 6]
```diff
@@ -1,359 +1,559 @@
...
-                timeout = [props.action];
-                debounceTimeouts.push(timeout);
-            } else {
-                clearTimeout(timeout[1]);
-            }
-
-            timeout[1] = setTimeout(function () {
-                dispatch(props.action);
-            }, props.wait);
+      const loadNbRecords = async (dispatch, options) => {
```
Comment (rohanshiva) [inline static/index.html:130]: `loadNbRecord`

File: static/index.html [hunk 7]
```diff
@@ -1,359 +1,559 @@
...
+      };
+      const CopyEmbed = (state, publicUrl) => {
+        const embedContent = embedNotebook(publicUrl);
+        navigator.clipboard.writeText(embedContent);
+        return [
+          { ...state, embed: "copied!" },
+          window.hyperappFx.Debounce({ wait: 3000, action: CopyAlert }),
+        ];
+      };
+      const DeleteNotebook = (state, delKey) => {
```
Comment (rohanshiva) [inline static/index.html:202]: `DeleteNotebook` action

File: static/index.html [hunk 8]
```diff
@@ -1,359 +1,559 @@
...
-                        name: payload.value
-                    }
-                }
+            active: newKey,
+          },
+          [renderIcons],
+        ];
+      };
+
+      const CreateWrapper = (state, first = false) => {
```
Comment (rohanshiva) [inline static/index.html:249]: `CreateWrapper`

File: static/index.html [hunk 9]
```diff
@@ -1,359 +1,559 @@
...
+        return [
+          {
+            ...state,
+          },
+          [renderIcons],
+          [updateNbRecord, { record: state.notebooks[state.active] }],
+        ];
+      };
+
+      const SaveCalculation = (state, event) => {
```
Comment (rohanshiva) [inline static/index.html:267]: `SaveCalculation` action

File: static/public.html (hunk 2 above)
Comment (rohanshiva) [reply static/public.html:80]: it returns the public note record
`public/:segment` returns the page, and `/public/raw/:segment` returns the note in a list like this 
`[{name: "notebook 1", content: "jdhdh", active: true, public: true}]`

**Pull request merged**